src/ast/value/datetime.rs

```diff
@@ -389,3 +389,92 @@ impl Iterator for DateTimeFieldIterator {
         self.0.clone()
     }
 }
+
+/// Similar to a [`DateTimeField`], but with a few more options
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum ExtractField {
+    Millenium,
+    Century,
+    Decade,
+    Year,
+    Quarter,
+    Month,
+    Day,
+    Hour,
+    Minute,
+    Second,
+    Millisecond,
+    Microsecond,
+    // Weirder fields
+    Timezone,
+    TimezoneHour,
+    TimezoneMinute,
+    WeekOfYear,
+    /// The day of the year (1 - 365/366)
+    DayOfYear,
```

Comment:
There are two more potential EXTRACT fields listed [here](https://www.postgresql.org/docs/12/functions-datetime.html#FUNCTIONS-DATETIME-EXTRACT): `isodow`, `isoyear`. Do we want to support those?

Reply:
oh yeah. Thanks!